types/cucumber/index.d.ts

```diff
@@ -77,7 +90,7 @@ export function Given(pattern: RegExp | string, options: StepDefinitionOptions,
 export function setDefaultTimeout(time: number): void;
 export function setDefinitionFunctionWrapper(fn: ((fn: () => void) => (...args: any[]) => any) | ((fn: () => void, options?: {[key: string]: any}) => (...args: any[]) => any)): void;
 // tslint:disable-next-line ban-types
-export function setWorldConstructor(world: ((this: World, init: {attach: Function, parameters: {[key: string]: any}}) => void) | {}): void;
+export function setWorldConstructor<T = World>(world: ((this: T, init: {attach: AttachmentFunction, parameters: WorldParameters}) => void) | { new (): T }): void;
```

Comment:
`T` should be removed in favour of `World` here too, just like all the others, because the intent of the type is (as far as I can tell) to allow (1) an init function from a World object or (2) a World constructor function. But `T` isn't used aside from that.

Reply:
I think `setWorldConstructor` accepts either a `constructor function` or a class.
but... anyway what should be the signature in the case you mentioned?

Reply:
to Typescript, a class `World` is just an object with `new(): World`, so you can pass either a constructor function or a class.

Here is what I would suggest:

```suggestion
export function setWorldConstructor(world: ((this: World, init: {attach: AttachmentFunction, parameters: WorldParameters}) => void) | { new (): World }): void;
```

This works the same because `CustomWorld` is a subtype of `World`, so it's fine to pass to `setWorldConstructor`. `setWorldConstructor` only cares that it receives *some* `World`, so it doesn't need `T` to capture a particular subtype of it.